tests/plugins/dut_monitor/dut_monitor.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import time
+import os
+
+from  datetime import datetime
+
+
+FETCH_CPU_CMD = "ps -eo pcpu,args | sort -k 1 -r | tail -n +2"
+FETCH_HDD_CMD = "df -hm /"
+MEASURE_DELAY = 2
+
+
+def process_cpu(log_file):
+    """
+    @summary: Fetch CPU utilization. Write to the file total and top 10 process CPU utilization.
+    @param log_file: Opened file object to store fetched CPU utilization.
+    """
+    buff = []
+    general_template = """
+"{timestamp}":
+    total: {total}
+    top_consumer:
```

Comment:
It looks like the log files are yaml format. Suggest to use dictionary like objects to hold collected information and then dump the dictionary object to string using the yaml lib. This is less error prune than manipulating string templates. The string dumped into the log file can be later loaded by yaml.safe_load().

Reply:
Depends on running time, dictionary can become huge. It will increase script performance and RAM utilization. In such approach text which is stored in the log file is already read by yaml library in the "pytest_dur_monitor.py":

"yaml.safe_load("".join(fp))"

Reply:
OK.